### PR TITLE
ci: fix codegen err

### LINKED ISSUE #7287: make codegen is not failing the CI action when changes are needed
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
`make codegen` in master currently leaves some files changed since #7264 was merged. The codegen CI action should have failed before the merge, but it did not identify the problem. When #7264 was backported, the CI failure was detected as expected in #7274. So the `make codegen` validation is only broken in master.

**Expected behavior:**
The specific codegen issue is fixed with #7286, but we still need to fix the CI action to catch future issues.

**How to reproduce it (minimal and precise):**

- Make a change to the codegen types
- Run `make codegen`
- See the changed file(s) locally
- If the generated file(s) are not included in the PR, we expect the CI action to fail

### PR DESCRIPTION
**Pull request by subhamkrai**
Look at individual commit message.

Signed-off-by: subhamkrai <srai@redhat.com>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #7287 

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.
[skip ci]

### CODE REVIEW

**Comment by subhamkrai** [comment]
I did minor changes in `types.go`(so the see codegen error) and also changes the `validate_modified_files.sh` similar to what we have in 1.5(reason: as codegen is working fine in 1.5). So, according to my understanding, I don't think there is a problem with the script. Debugging more

File: .github/workflows/codegen.yml [hunk 1]
```diff
@@ -8,13 +8,15 @@ jobs:
...
       with:
         go-version: 1.15
 
+    - name: checkout
+      uses: actions/checkout@v2
+      with:
+        path: $(go env GOPATH)/src
+
     - name: run codegen
       run: GOPATH=$(go env GOPATH) make codegen
```
Comment (leseb) [inline .github/workflows/codegen.yml:21]: I think we need to change dir; see `working-directory`

Comment (subhamkrai) [reply .github/workflows/codegen.yml]: `Working directory is '/home/runner/work/rook/rook/$(go env GOPATH)/src'`... I change manually to point `"/home/runner/go/src"` but no luck do you see what I'm missing. @leseb

Comment (subhamkrai) [reply .github/workflows/codegen.yml]: ahh. it worked...now I see. Thanks @leseb

File: .github/workflows/codegen.yml [hunk 2]
```diff
@@ -8,16 +8,19 @@ jobs:
...
+    - name: copy working directory to GOPATH
+      run: sudo mkdir -p /home/runner/go/src/github.com && sudo cp -a /home/runner/work/rook /home/runner/go/src/github.com/
+
     - name: run codegen
-      run: GOPATH=$(go env GOPATH) make codegen
+      run: cd /home/runner/go/src/github.com/rook/rook && GOPATH=$(go env GOPATH)/ make codegen
 
     - name: validate codegen
-      run: tests/scripts/validate_modified_files.sh codegen
+      run: cd /home/runner/go/src/github.com/rook/rook && tests/scripts/validate_modified_files.sh codegen
```
Comment (leseb) [inline .github/workflows/codegen.yml:25]: Why not use `working-directory` here?

File: .github/workflows/codegen.yml [hunk 3]
```diff
@@ -8,16 +8,19 @@ jobs:
...
 
+    - name: checkout
+      uses: actions/checkout@v2
+
+    - name: copy working directory to GOPATH
+      run: sudo mkdir -p /home/runner/go/src/github.com && sudo cp -a /home/runner/work/rook /home/runner/go/src/github.com/
+
     - name: run codegen
-      run: GOPATH=$(go env GOPATH) make codegen
+      run: cd /home/runner/go/src/github.com/rook/rook && GOPATH=$(go env GOPATH)/ make codegen
```
Comment (leseb) [inline .github/workflows/codegen.yml:22]: Why not use `working-directory` here?

Comment (subhamkrai) [reply .github/workflows/codegen.yml]: Done.

**Review by leseb — APPROVED** [review]

**Pull request merged**